text/0401-coercions.md

```diff
@@ -321,18 +321,26 @@ following holds:
 
 * `e` has type `T` and `T` coerces to `U`;
 
-* `e` has type `*T` and `U` is `*U_0` (i.e., between any raw pointers);
+* `e` has type `*T`, `U` is `*U_0`, and either `U_0: Sized` or
+    unsize_kind(`T`) = unsize_kind(`U_0`);
 
-* `e` has type `*T` and `U` is `uint` , or vice versa;
+* `e` has type `*T` and `U` is `usize`, or vice versa, while `T: Sized`;
```

Comment:
I think we can also allow casts from *T to usize where T is unsized (it's not clear if that is what you meant or not)